### PR TITLE
planner: refactor some code of binding cache 

### PR DESCRIPTION
**Pull request by qw4990**
### What problem does this PR solve?

Issue Number: ref #51347

Problem Summary: planner: refactor some code of binding cache

### What changed and how does it work?

planner: refactor some code of binding cache

### Check List

Tests 

- [ ] Unit test
- [x] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: pkg/planner/optimize.go [hunk 1]
```diff
@@ -241,32 +241,13 @@ func Optimize(ctx context.Context, sctx sessionctx.Context, node *resolve.NodeW,
...
-	var bindings bindinfo.Bindings
+	binding, match, _ := bindinfo.MatchSQLBinding(sctx, stmtNode)
+	var bindings []*bindinfo.Binding
 	if match {
-		bindings = []bindinfo.Binding{binding}
+		bindings = []*bindinfo.Binding{binding}
 	}
 
 	useBinding := enableUseBinding && isStmtNode && match
-	if sessVars.StmtCtx.EnableOptimizerDebugTrace {
```
Comment (qw4990) [inline pkg/planner/optimize.go:251]: Remove to keep the code simple since no one is using OptimizerDebugTrace.

**Review by AilinKid — APPROVED** [review]

**Review by lance6716 — COMMENTED** [review]
/approve

for domain part

**Review by winoros — APPROVED** [review]

**Review by hawkingrei — COMMENTED** [review]
LGTM

**Review by hawkingrei — COMMENTED** [review]
LGTM, although it has been merged.

**Pull request merged**